packages/app/src/lib/audit.ts

```diff
@@ -0,0 +1,241 @@
+                const isThisPasswordWeak = await isPasswordWeak(passwordField.field.value);
+                if (isThisPasswordWeak) {
+                    // Don't add the same item twice to the list, if there are more than one weak password fields in it
+                    if (!weakPasswordItemIds.has(item.id)) {
+                        weakPasswords.push({ item, vault });
+                        weakPasswordItemIds.add(item.id);
+                    }
+
+                    auditResults.push({
+                        type: AuditResultType.WeakPassword,
+                        fieldIndex: passwordField.fieldIndex,
+                    });
+
+                    vaultResultsFound = true;
+                }
+
+                // Perform compromised audit
+                const isPasswordCompromised = await hasPasswordBeenCompromised(passwordHash);
+                if (isPasswordCompromised) {
+                    // Don't add the same item twice to the list, if there are more than one compromised password fields in it
+                    if (!compromisedPasswordItemIds.has(item.id)) {
+                        compromisedPasswords.push({ item, vault });
+                        compromisedPasswordItemIds.add(item.id);
+                    }
+
+                    auditResults.push({
+                        type: AuditResultType.CompromisedPassword,
+                        fieldIndex: passwordField.fieldIndex,
+                    });
+
+                    vaultResultsFound = true;
+                }
+            }
+
+            item.auditResults = auditResults;
+            item.lastAudited = new Date();
+            vault.items.update(item);
+        }
+
+        if (vaultResultsFound) {
+            await app.saveVault(vault);
+        }
+
+        resultsFound = resultsFound || vaultResultsFound;
+    }
+
+    if (resultsFound) {
+        await app.save();
+    }
+
+    return {
+        reusedPasswords,
+        weakPasswords,
+        compromisedPasswords,
+    };
+}
+
+export function noItemsTextForAudit(type: AuditResultType) {
+    switch (type) {
+        case AuditResultType.WeakPassword:
+            return $l("You don't have any items with weak passwords!");
+        case AuditResultType.ReusedPassword:
+            return $l("You don't have any items with reused passwords!");
+        case AuditResultType.CompromisedPassword:
+            return $l("You don't have any items with compromised passwords!");
+        default:
+            return $l("You don't have any insecure items!");
+    }
+}
+
+export function titleTextForAudit(type: AuditResultType) {
+    switch (type) {
+        case AuditResultType.WeakPassword:
+            return $l("Weak Passwords");
+        case AuditResultType.ReusedPassword:
+            return $l("Reused Passwords");
+        case AuditResultType.CompromisedPassword:
+            return $l("Compromised Passwords");
+        default:
+            return $l("Insecure");
+    }
+}
```

Comment:
Moved these two functions here along with the `auditVaults` function so they can easily be imported and used throughout components.

Reply:
Also, for this kind of thing I prefer switch/case, simply because it looks nicer :)